src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/GroupResponsibilityController.kt

```diff
@@ -37,17 +37,14 @@ class GroupResponsibilityController(
         return modellingGroupRepo.getTouchstonesByGroupId(groupId).filterByPermission(context)
     }
 
-    fun getResponsibilities(): ResponsibilitySet
+    fun getResponsibilities(): ResponsibilitySetWithExpectations
     {
         val groupId = groupId(context)
         val touchstoneVersionId = context.params(":touchstone-version-id")
         val filterParameters = ScenarioFilterParameters.fromContext(context)
-
-        modellingGroupRepo.getModellingGroup(groupId)
-        val data = responsibilitiesRepo
-                .getResponsibilitiesForGroupAndTouchstone(groupId, touchstoneVersionId, filterParameters)
-        context.checkIsAllowedToSeeTouchstone(touchstoneVersionId, data.touchstoneStatus)
-        return data.responsibilitySet
+        val (responsibilities, touchstoneStatus) = expectationsLogic.getResponsibilitySetWithExpectations(groupId, touchstoneVersionId, filterParameters)
```

Comment:
I know we were doing this before, but it seems kind of hack-y to be calling a method called `getResponsibilitySetWithExpectations` and actually getting back the touchstone status as a sort of bolt on. 

Can we just make a separate request to get the touchstone status and have this new logic method do what it says on the tin? From what I can see the underlying `ResponsibilitiesRepository` method just calls `touchstoneRepository.touchstoneVersions.get(touchstoneVersionId)` to get the status and pass it back. We may as well just call that directly here.

Reply:
Yes you're right. I tried to figure out how to simplify this before, but got confused. This is much better. Note that there is now an extra webmodels PR: https://github.com/vimc/montagu-webmodels/pull/57